apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -50,6 +52,44 @@
     this.logger.debug('Clock updated', now);
   };
 
+	LockScreenClockWidget.prototype.updateAlarm =
+	function() {
+    var self = this;
+		// returns alarm info from DataStore
+		navigator.getDataStores('alarms')
+			.then( function(stores){
+        stores[0].getLength().then(function(len){
+          if(len > 0) {
+            stores[0].get(1)
+              .then( function(data) {
+                if(data.data) {
+                  if(data.data.hour && data.data.minute) {
```

Comment:
And also it's very confused to see code like

    if (data.data.....)

Could you please change it as

    if (fetchingResult.data)

or something like that?